### PR TITLE
CGUIFontCache: use std::chrono and some other changes

### PR DESCRIPTION
**Pull request by lrusak**
I wanted to separate this from #19808 as it's a bit more involved and includes some other changes.

1) use `std::chrono::steady_clock::time_point` for the font cache check. The old implementation used `SystemClockMillis` which was just a wrapper anyways.
2) ~~directly initialize the cache item from the key. I'm not 100% sure about this one as it seems the old way would use a pointer to point to the key item, this update simplifies the logic but will make a copy (like the other items do already?). If desired I can drop this change.~~
3) a small safety update to use a `unique_ptr`
4) another saftey update to use a `unique_ptr`

clang-format did some heavy adjustments so it's likely easier looking at each commit individually.

### CODE REVIEW

File: xbmc/guilib/GUIFontCache.cpp [hunk 1]
```diff
@@ -12,14 +12,21 @@
 #include <stdint.h>
 #include <vector>
 
+using namespace std::chrono_literals;
+
+namespace
+{
+constexpr auto FONT_CACHE_TIME_LIMIT = 1000ms;
```
Comment (Montellese) [inline xbmc/guilib/GUIFontCache.cpp:19]: `static`

File: xbmc/guilib/GUIFontCache.h [hunk 2]
```diff
@@ -134,7 +134,7 @@ struct CGUIFontCacheKeysMatch
 template<class Position, class Value>
 class CGUIFontCache
 {
-  CGUIFontCacheImpl<Position, Value>* m_impl;
+  std::unique_ptr<CGUIFontCacheImpl<Position, Value>> m_impl;
```
Comment (Montellese) [inline xbmc/guilib/GUIFontCache.h:137]: Does this really work? While using `std::unique_ptr<>` with incomplete types (forward declared) is allowed it usually results in nasty compiler errors if the type's destructor is inline (or not defined at all).

Comment (garbear) [reply xbmc/guilib/GUIFontCache.h:137]: If `CGUIFontCacheImpl` has a cpp file then the destructor could be defined there, avoiding those nasty errors

File: xbmc/guilib/GUIFontCache.cpp (hunk 1 above)
Comment (yol) [reply xbmc/guilib/GUIFontCache.cpp:19]: @Montellese anonymous namespaces are equivalent to static

Comment (Montellese) [reply xbmc/guilib/GUIFontCache.cpp:19]: You're right, I missed the anonymous namespace.

File: xbmc/guilib/GUIFontCache.cpp [hunk 3]
```diff
@@ -154,7 +171,11 @@ Value &CGUIFontCacheImpl<Position, Value>::Lookup(Position &pos,
     // Cache miss
     dirtyCache = true;
     CGUIFontCacheEntry<Position, Value> *entry = nullptr;
-    if (!m_list.ageMap.empty() && (nowMillis - m_list.ageMap.begin()->first) > FONT_CACHE_TIME_LIMIT)
+
+    const auto duration =
+        std::chrono::duration_cast<std::chrono::milliseconds>(now - m_list.ageMap.begin()->first);
```
Comment (yol) [inline xbmc/guilib/GUIFontCache.cpp:169]: you can't execute `m_list.ageMap.begin()->` if `m_list.ageMap` is empty

File: xbmc/guilib/GUIFontCache.h [hunk 4]
```diff
@@ -37,25 +37,33 @@ template<class Position>
...
   Position m_pos;
-  std::vector<UTILS::Color> &m_colors;
-  vecText &m_text;
+  std::vector<UTILS::Color> m_colors;
+  vecText m_text;
   uint32_t m_alignment;
   float m_maxPixelWidth;
   bool m_scrolling;
-  const TransformMatrix &m_matrix;
+  const TransformMatrix m_matrix;
```
Comment (yol) [inline xbmc/guilib/GUIFontCache.h:45]: why remove this reference?

**Review by yol — CHANGES_REQUESTED** [review]

File: xbmc/guilib/GUIFontCache.h (hunk 4 above)
Comment (lrusak) [reply xbmc/guilib/GUIFontCache.h]: good catch

File: xbmc/guilib/GUIFontCache.cpp (hunk 3 above)
Comment (lrusak) [reply xbmc/guilib/GUIFontCache.cpp]: Thanks! fixed.

**Review by yol — APPROVED** [review]

**Comment by lrusak** [comment]
I've rebased these changes now that the other font changes have gone in. @AlwinEsch can you take a look please?

**Review by AlwinEsch — APPROVED** [review]
From overview it looks good and better.

**Pull request merged**